src/scripts/migrate.py

```diff
@@ -91,30 +118,14 @@ def slurp(
     owned_terms: List[Term] = _get_all_owned_terms(
         ontology=ontology, owned_prefix_map=owned_prefix_map, ontology_path=ontology_path, cache_dir_path=CACHE_DIR,
         onto_config_path=onto_config_path, use_cache=use_cache)
-    owned_term_curies: List[CURIE] = [x.curie for x in owned_terms]
     sssom_object_ids: Set[Union[URI, CURIE]] = set(sssom_df['object_id'])  # Usually CURIE, but spec allows URI
     unmapped_terms: List[Term] = [x for x in owned_terms if x.curie not in sssom_object_ids]
     slurp_candidates = [x for x in unmapped_terms if x.curie not in excluded_terms]  # remove exclusions
 
-    # Determine slurpable terms
-    # Migratable term: A term owned by the given ontology but unmapped in Mondo, which has either no parents, or if it
-    # has parents, all of its parents have already been mapped in Mondo.
+    # Determine slurpable / migratable terms
     terms_to_slurp: List[Dict[str, str]] = []
     for t in slurp_candidates:
-        # If all T.parents mapped, and 1+ is an exact or narrow match and non obsolete, designate T for slurping
-        # (i.e. only slurp if parents are already slurped)
-        qualified_parent_mondo_ids = []
-        no_parents: bool = not t.direct_owned_parent_curies
-        for parent_curie in t.direct_owned_parent_curies:
-            # Conversely, if any of T.parents is unmapped, T is not to be slurped
-            if parent_curie in sssom_object_ids:
-                parent: Dict = sssom_df[sssom_df['object_id'] == parent_curie].to_dict('records')[0]
-                parent_mondo_id, parent_mondo_label = parent['subject_id'], parent['subject_label']
-                if not parent_mondo_label.startswith('obsolete'):
-                    qualified_parent_mondo_ids.append(parent_mondo_id)
-            if parent_curie not in sssom_object_ids and parent_curie in owned_term_curies:
-                break
-        if qualified_parent_mondo_ids or no_parents:
+        if not parent_conditions_on or _check_parent_conditions(t, sssom_object_ids, sssom_df):
```

Comment:
@matentzn Implemented new default logic where we do not check anything about a term's parents when determining if it is migratable. Otherwise, there is a function called `_check_parent_conditions()` where it will include that logic if we pass the `--parent-conditions-on` param. 

I was actually going to name this `_is_migration_candidate()` as you proposed, but if you look a few lines above, you will see that we're already iterating over slurp candidates: `    for t in slurp_candidates:`. This is because earlier in the script we've already filtered out any terms that are already mapped or excluded.

Reply:
I think your suggestion of doing (b), the refactor, makes sense, but please do not drop the parent check, and also note my comment on the exactMatch/narrowMatch condition.

The logic here has become a bit too complex for nico-style review. Try to focus on understanding what the outcome of a feature like this one should be, and then lets communicate through bug reports rather than implementation details - I don't know whats best, and our dialogue is likely misleading you many times. 

Finish the PR in whatever way you see fit, and show me the migration tables created, and we discuss it from there!

Reply:
Hey Nico, yes, I agree, this is getting too confusing. I think that a meeting is the most efficient thing if there are a lot of complexities or ambiguities.

I only have one question left on something unclear but has only a minimal impact, if any, so I'll save it for our next meeting.

I'll do the following things now:
i. Implement my simplification refactor.
ii. Reactivate the parent conditions, but change the logic a little as you [just described](https://github.com/monarch-initiative/mondo-ingest/pull/316#discussion_r1199793432), _all parents should either be (a) mapped, (b) obsolete, or (c, new) excluded_
iii. Move the exact/narrow match conditions out of 'ii', valid parent conditions, and add it [instead](https://github.com/monarch-initiative/mondo-ingest/pull/316#discussion_r1199794773) to the list of parents that show up in the TSV

Reply:
@matentzn I've just finished the logic updates and run it for 3 ontologies. It's still running so I'll commit the others when it's done.

Can you merge this after face checking the outputs if they're good?

Perhaps you're right about it not being a good use of your time to check on implementation, however. I have an idea.
Rather than waiting for bug reports, this is a good opportunity to implement some testing. Sabrina could probably come up with good cases, but you would likely be better at coming up with potential tricky edge cases. It would be nice if I could get a list of terms from you guys, and then for each of those, whether or not that term would be expected in the slurp output.